src/main/java/com/google/sps/model/CalendarClientImpl.java

```diff
@@ -60,6 +61,29 @@ private CalendarClientImpl(Credential credential) {
     return events != null ? events : new ArrayList<>();
   }
 
+  @Override
+  public List<Event> getUpcomingEvents(
+      CalendarListEntry calendarList, DateTime timeMin, DateTime timeMax) throws IOException {
```

Comment:
Suggestion: Have you taken a look at [Date](https://docs.oracle.com/javase/8/docs/api/java/util/Date.html)? I think that might be slightly more convenient to work with. Particularly, that allows you to compare them using methods like `before()` and `after()`.

Reply:
The calendar API gives DateTime types and requires DateTime types to get and set Events.